### ISSUE TITLE
the calculation of exploitability

### ISSUE THREAD

**Issue by wweichn** [DoktorDaveJoos/Neural-Ficititious-Self-Play-in-Imperfect-Information-Games#1]
when I run your code, I can't get a consistent decreasing trend of exploitability. And I found you calculate exploitability by calculating Q value, I don't think its right...
So I wonder how do you get the graph in your thesis. Do I miss something?thanks

**Comment by DoktorDaveJoos** [comment]
You are absolutely right, in this version of my thesis - the calculation of the q values is noisy and biased.  Noisy because it takes samples during play. Biased because it computes it from an imperfect, approximate BR. Leduc is a small game allowing exhaustive computation of exploitability via dynamic
programming in the whole game tree. In particular, to compute exploitability of player 1's avg policy. -Construct the whole game tree with all decisions of player 1 following his avg policy, i.e. fixed. -Use
dynamic programming starting from the leafs to recursively compute the best decision at each node of player 2, also backing up the values -This exhaustively computes a perfect best response for player 2, and you get the expected value of this BR from the values you've backed up. [This approach is non-specific to where the avg policy came from.]
 Cheers,
David

**Comment by jason-scott-20** [comment]
So how do you get the graph  in your thesis? Was it NOT created with NFSP?
